### ISSUE TITLE
[TEST] Harvester v1.2.2 - Rancher v2.7.13 Integration Test

### ISSUE THREAD

**Issue by albinsun** [harvester/tests#1280]
## Prerequisites
1. Setup 3 nodes Harvester `v1.2.2` (QEMU/KVM, 8C/16G/500G per node)
1. Create a new storage class apart from default one.
1. Create 2 networks on different NICs: `mgmt-vlan1` and `bond-vlan1`
1. Create 2 VMs with extra disk and data with md5sum: 1 running and 1 stopped
1. Take VM backups and snapshots on 2 VMs 

### `v2.7.13`
1. Setup Rancher `v2.7.13` (Helm with `v1.27.13+k3s1` on 2C/4G QEMU/KVM)
2. Import Harvester to Rancher
1. Create single node **RKE2** cluster
   * `harvester_e2e_tests/integrations/test_9_rancher_integration.py`
   * Scale up to 2 nodes
   * Scale down to 1 node
1. Create single node **RKE1** cluster
   * `harvester_e2e_tests/integrations/test_9_rancher_integration.py`
   * Scale up to 2 nodes
   * Scale down to 1 node
1. Create single node **RKE2** cluster via Terraform
   * `harvester_e2e_tests/integrations/test_z_terraform_rancher.py`

### `v2.7.12` -> `v2.7.13`
1. Setup Rancher `v2.7.12` (Helm with `v1.27.13+k3s1` on 2C/4G QEMU/KVM)
1. Import Harvester to Rancher
1. Create single node **RKE2** cluster
   * `harvester_e2e_tests/integrations/test_9_rancher_integration.py`
1. Upgrade Rancher to `v2.7.13`
   * Cluster still `Active`
   * Nginx deployment and LB still works
   * Scale up to 2 nodes
   * Scale down to 1 node

## Reference
1. harvester/tests/issues/1150
1. harvester/tests/issues/900
1. harvester/tests/issues/1279
2. https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-7-13/
3. https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/install-upgrade-on-a-kubernetes-cluster/upgrades#steps-to-upgrade-rancher

**Comment by albinsun** [comment]
## Prerequisites
1. <details>
   <summary> Setup 3 nodes Harvester <code>v1.2.2</code> (QEMU/KVM, 8C/16G/500G per node) :heavy_check_mark: 
   </summary>
   
   ![image](https://github.com/harvester/tests/assets/2773781/73fded64-fc8d-48d3-8a67-ccaa753f6aff)
   </details>

1. <details>
   <summary> Create a new storage class apart from default one. :heavy_check_mark: 
   </summary>
   
   ![image](https://github.com/harvester/tests/assets/2773781/4369530f-b46f-4dda-b93c-d4a93c1d227f)
   </details>

1. <details>
   <summary> Create 2 networks on different NICs: <code>mgmt-vlan1</code> and <code>bond-vlan1</code> :heavy_check_mark: 
   </summary>
   
   ![image](https://github.com/harvester/tests/assets/2773781/652484b9-a9ea-42e7-8954-27a05f8d40ca)
   </details>

1. Create 2 VMs with extra disk and data with md5sum: 1 `Running` and 1 `Off` :heavy_check_mark: 
   * <details>
     <summary> running w/ <code>bond-vlan1</code>
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/a2b9c453-f98d-44d1-90b4-1501ce53eddd)
     </details>

   * <details>
     <summary> stopped w/ <code>mgmt-vlan1</code>
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/b1bbf0c2-ebc3-40ac-8357-3923f135edb3)
     </details>

   * <details>
     <summary> Both with extra data and md5sum
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/e3108e1f-c644-48c7-a5eb-a440f32d7fe9)
     </details>

1. Take VM backups and snapshots on 2 VMs :heavy_check_mark: 
   * <details>
     <summary> Backup
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/9371ac9b-8217-459a-9d1b-b4bfae724ab2)
     </details>
     
     > Not hit known harvester/harvester/issues/4358
     
   * <details>
     <summary> Snapshot
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/c8514180-7aba-4712-b2d9-be19e81ebbec)
     </details>

**Comment by albinsun** [comment]
## Rancher Integration - `v2.7.13`
1. <details>
   <summary> Setup Rancher <code>v2.7.13</code> (Helm with <code>v1.27.13 +k3s1</code> on 2C/4G QEMU/KVM) :heavy_check_mark: 
   </summary>
   
   ![image](https://github.com/harvester/tests/assets/2773781/c2bcebbd-8c3e-4423-a143-19b5c630910d)
   </details>

1. Create single node **RKE2** cluster (`v1.27.13+rke2r1`) :heavy_check_mark: 
   * <details>
     <summary> <code>harvester_e2e_tests/integrations/test_9_rancher_integration.py::TestRKE2</code>
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/f83ba08b-8ddf-4da8-bdc0-56e694467210)
     ![image](https://github.com/harvester/tests/assets/2773781/87167fbc-6f30-48ed-b4ba-01ce4a93668b)
     </details>

     * `test_import_harvester`
     * `test_create_rke2`
     * `test_create_pvc`
     * `test_csi_deployment`
     * `test_delete_deploymen`
     * `test_deploy_nginx`
     * `test_load_balancer_service[1-dhcp]`
     * `test_load_balancer_service[1-pool]`
     
   * <details>
     <summary> Scale up to 2 nodes
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/dff0a919-5a93-45df-a192-280290a2d496)
     ![image](https://github.com/harvester/tests/assets/2773781/a21f1bc0-fedc-48e2-b408-7a8e2ae401fc)
     Harvester
     ![image](https://github.com/harvester/tests/assets/2773781/05589a95-d7ad-46d5-bcd4-108a099edbbb)
     </details>
     
   * <details>
     <summary> Scale down to 1 node
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/69f5c440-63d1-42e9-b88d-2b276df48112)
     </details>

1. Create single node **RKE1** cluster (`v1.27.13-rancher1-1`) :heavy_check_mark: 
   * <details>
     <summary> <code>harvester_e2e_tests/integrations/test_9_rancher_integration.py::TestRKE1</code>
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/058eda1f-b99a-44ee-bd2e-6b8d2ca8a724)
     ![image](https://github.com/harvester/tests/assets/2773781/9b67330e-fadb-4e9a-9783-284c04679eee)
     Harvester
     ![image](https://github.com/harvester/tests/assets/2773781/405eba9d-e1ae-4ee7-ad32-9eefb644b004)
     </details>
     
     * `test_import_harvester`
     * `test_create_rke1`
     * `test_create_pvc`
     * `test_cloud_provider_chart`
     * `test_deploy_nginx`
     * `test_load_balancer_service[1-dhcp]`
     * `test_load_balancer_service[1-pool]`
     * `test_csi_driver_chart`
     * `test_csi_deployment`
     * `test_delete_deployment`
     
   * <details>
     <summary> Scale up to 2 nodes
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/fe1e0d62-9e46-4c42-a139-c6dc6696465e)
     ![image](https://github.com/harvester/tests/assets/2773781/ef6c03c4-90c3-4980-81fa-9aa1a0303d34)
     Harvester
     ![image](https://github.com/harvester/tests/assets/2773781/20372ac8-02c5-4224-b953-a153f624ff67)
     </details>
     
   * <details>
     <summary> Scale down to 1 node
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/2580dac8-e756-46f4-8ad5-758106af31ca)
     ![image](https://github.com/harvester/tests/assets/2773781/4bc5ffef-ed41-40d2-9bdf-1834f7db0444)
     Harvester
     ![image](https://github.com/harvester/tests/assets/2773781/13733c1d-cd59-46d2-86aa-d2c585f16553)
     </details>

1. Create single node **RKE2** cluster via Terraform :heavy_check_mark: 
   * `terraform version`
     ```
     Terraform v1.8.4
     on linux_amd64
     + provider registry.terraform.io/rancher/rancher2 v3.1.1
     ```
   * <details>
     <summary> <code>harvester_e2e_tests/integrations/test_z_terraform_rancher.py</code>
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/f5e37531-4783-4347-837f-8995d7712698)
     </details>
     
     * `test_import_harvester`
     * `test_create_cloud_credential`
     * `test_create_machine_config`
     * `test_create_rke2_cluster`
     * `test_delete_rke2_cluster`
     * `test_delete_machine_config`
     * `test_delete_cloud_credential`

**Comment by albinsun** [comment]
## Rancher Integration - `v2.7.12` -> `v2.7.13`
1. <details>
   <summary> Setup Rancher <code>v2.7.12</code> (Helm with <code>v1.27.13 +k3s1</code> on 2C/4G QEMU/KVM) :heavy_check_mark: 
   </summary>

   ![image](https://github.com/harvester/tests/assets/2773781/a10a4743-81b4-426b-86e8-daa71deeddcf)
   </details>
   
1. Single node **RKE2** cluster (`v1.27.13+rke2r1`) :heavy_check_mark: 
   * <details>
     <summary> <code>harvester_e2e_tests/integrations/test_9_rancher_integration.py::TestRKE2</code>
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/455518d2-bb37-48b5-b519-853d9a5af675)
     ![image](https://github.com/harvester/tests/assets/2773781/8f565285-e307-4118-ab5f-488cb73b4233)
     Harvester
     ![image](https://github.com/harvester/tests/assets/2773781/a14e8dc4-51ab-4aab-997a-92e50821133d)
     </details>
     
     * `test_import_harvester`
     * `test_create_rke2`
     * `test_create_pvc`
     * `test_csi_deployment`
     * `test_delete_deployment`
     * `test_deploy_nginx`
     * `test_load_balancer_service[1-dhcp]`
     * `test_load_balancer_service[1-pool]`
 
1. Upgrade Rancher to `v2.7.13` :heavy_check_mark: 
   * <details>
     <summary> Upgrade successfully
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/7b98d4eb-1cfd-4862-8724-c49e3de9efa2)
     </details>

   * <details>
     <summary> Cluster still <code>Active</code>
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/98770f8d-8018-47a1-b452-4a8b3097fd7c)
     Harvester
     ![image](https://github.com/harvester/tests/assets/2773781/af8246fb-2e37-407e-ab01-d8a34bed3737)
     </details>
     
   * <details>
     <summary> Nginx deployment and LB still works
     </summary>
     
     Deployment
     ![image](https://github.com/harvester/tests/assets/2773781/e6c2ec73-a5e6-458e-81a8-aa36e59b61d7)
     LB
     ![image](https://github.com/harvester/tests/assets/2773781/37fbe60e-dc15-4067-99d0-078cd6dc829f)
     Harvester
     ![image](https://github.com/harvester/tests/assets/2773781/017e9726-048d-4fb5-b2af-b3de1b5cb1a7)
     </details>

   * <details>
     <summary> Scale up to 2 nodes
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/a986c6ad-f907-49d1-8e18-d0c6d741d337)
     ![image](https://github.com/harvester/tests/assets/2773781/64d66a5b-931c-4803-85ff-b6fd236320dd)
     Harvester
     ![image](https://github.com/harvester/tests/assets/2773781/afb690a5-70cf-4c64-ada6-504320d8ddc5)
     </details>
     
   * <details>
     <summary> Scale down to 1 node
     </summary>
     
     ![image](https://github.com/harvester/tests/assets/2773781/b2927891-e63b-4784-b96e-cad65ee9e314)
     ![image](https://github.com/harvester/tests/assets/2773781/bc24787f-32e2-4ced-ade1-b1d68932e370)
     Harvester
     ![image](https://github.com/harvester/tests/assets/2773781/4b57987a-ffba-41a3-bc77-ca5c8cb69c07)

     </details>

**Comment by albinsun** [comment]
Provide test result on `harvester-v1.2.2` with
1. `rancher-v2.7.13` ([detail](https://github.com/harvester/tests/issues/1280#issuecomment-2128482523))
2. `rancher-v2.7.12` -> `rancher-v2.7.13` -> `rancher-v2.8.4` ([detail](https://github.com/harvester/tests/issues/1280#issuecomment-2128668721))

Issues hit: 
1. Known harvester/harvester/issues/4358
   * Hit in `rancher-v2.8.4` but not in `v2.7.12` and `v2.7.13`
   * Source issue is planned to fix in `rancher-v2.9`
     rancher/rancher/issues/42582

cc. @khushboo-rancher
